lib/compress/zstd_compress.c

```diff
@@ -2805,6 +2813,11 @@ static size_t ZSTD_loadDictionaryContent(ZSTD_matchState_t* ms,
 
         ZSTD_overflowCorrectIfNeeded(ms, ws, params, ip, ichunk);
 
+        if (params->ldmParams.enableLdm && ls != NULL && srcSize >= params->ldmParams.minMatchLength) {
+            ZSTD_ldm_fillHashTable(ls, (const BYTE*)src, (const BYTE*)src + srcSize, &params->ldmParams);
+            break;
```

Comment:
Why break here? This stops the dictionary from filling non-LDM tables. Could this be why compression ratio is hurt so much in the optimal parser when LDM is enabled?

Reply:
Yeah good catch. Not sure why I had this originally honestly. Although, even with it removed, the compression ratio doesn't change. Tested with llvm and lz4. Still, I'm gong to remove it because it doesn't make sense to not fill the other tables

Reply:
Ah alright, found a very small difference in compression ratio when I compress with --single-thread on wordpress

Reply:
Looks like there are small differences in file size that get exposed when compressing with --single-thread

Reply:
> when compressing with --single-thread

For my understanding, do you imply there is no difference _without_ `--single-thread` ?

Also, what's the status of `--single-thread` with regards to `--patch-from` btw ? Is it orthogonal ? It it implied ? etc.